src/topp/OpenSwathDecoyGenerator.cpp

```diff
@@ -162,33 +211,86 @@ class TOPPOpenSwathDecoyGenerator
       allowed_fragment_charges.push_back(charge);
     }
 
-    if (method != "shuffle" && method != "pseudo-reverse" && method != "reverse" && method != "shift")
-    {
-      throw Exception::IllegalArgument(__FILE__, __LINE__, OPENMS_PRETTY_FUNCTION, "No valid decoy generation method selected!");
-    }
-
-    TraMLFile traml;
     TargetedExperiment targeted_exp;
     TargetedExperiment targeted_decoy;
 
-    std::cout << "Loading " << in << std::endl;
-    traml.load(in, targeted_exp);
+    // Load data
+    LOG_INFO << "Loading targets from file: " << in << std::endl;
+    if (in_type == FileTypes::TSV || in_type == FileTypes::MRM)
+    {
+      const char* tr_file = in.c_str();
+      Param reader_parameters = getParam_().copy("algorithm:", true);
+      TransitionTSVReader tsv_reader = TransitionTSVReader();
+      tsv_reader.setLogType(log_type_);
+      tsv_reader.setParameters(reader_parameters);
+      tsv_reader.convertTSVToTargetedExperiment(tr_file, in_type, targeted_exp);
+      tsv_reader.validateTargetedExperiment(targeted_exp);
+    }
+    else if (in_type == FileTypes::PQP)
+    {
+      const char* tr_file = in.c_str();
+      TransitionPQPReader pqp_reader = TransitionPQPReader();
+      Param reader_parameters = getParam_().copy("algorithm:", true);
+      pqp_reader.setLogType(log_type_);
+      pqp_reader.setParameters(reader_parameters);
+      pqp_reader.convertPQPToTargetedExperiment(tr_file, targeted_exp);
+      pqp_reader.validateTargetedExperiment(targeted_exp);
+    }
+    else if (in_type == FileTypes::TRAML)
+    {
+      TraMLFile traml;
+      traml.load(in, targeted_exp);
+    }
 
     MRMDecoy decoys = MRMDecoy();
+    decoys.setLogType(ProgressLogger::CMD);
 
-    std::cout << "Generate decoys" << std::endl;
-    decoys.generateDecoys(targeted_exp, targeted_decoy, method, decoy_tag, identity_threshold, max_attempts, mz_threshold, mz_shift, exclude_similar, similarity_threshold, remove_CNterm_mods, precursor_mass_shift, allowed_fragment_types, allowed_fragment_charges, enable_detection_specific_losses, enable_detection_unspecific_losses, remove_unannotated);
+    LOG_INFO << "Generate decoys" << std::endl;
+    decoys.generateDecoys(targeted_exp, targeted_decoy, method, decoy_tag, identity_threshold, max_attempts, product_mz_threshold, product_mz_shift, similarity_threshold, precursor_mz_shift, allowed_fragment_types, allowed_fragment_charges, enable_detection_specific_losses, enable_detection_unspecific_losses);
 
-    if (append)
```

Comment:
We check that there is a minimum fraction of decoys vs targets that survive the filtering. With the default parameters and the PHL, about 95% of the targets are covered.